reconcile/openshift_base.py

```diff
@@ -345,11 +345,47 @@ def apply(
         except FieldIsImmutableError:
             # Add more resources types to the list when you're
             # sure they're safe.
-            if resource_type not in ["Route", "Service", "Secret"]:
+            if resource_type not in ["Route", "Service", "Secret", "Deployment"]:
```

Comment:
would it make sense to raise a different exception and handle this in a separate section?

Reply:
@maorfr now ok for you?

Reply:
Yes! Thank you!